### ISSUE TITLE
Tests fail in Fedora due to broken pyparsing compatibility

### ISSUE THREAD

**Issue by susilehtola** [MRChemSoft/mrchem#427]
Updating to mrchem 1.1.0 results in test failures on Fedora.
```
Test project /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build
      Start 29: HF_molecular_gradient_LDA
 1/30 Test #29: HF_molecular_gradient_LDA ........***Failed    0.64 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/hf_grad_lda/hf --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/hf_grad_lda/hf and args None

stdout


stderr
Traceback (most recent call last):
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 90, in <module>
    main()
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 55, in main
    user_dict = parse(infile=inp_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 108, in parse
    ir = lex(infile=infile, ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 40, in lex
    ir = lexer.lex_from_str(in_str=f.read(), ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/lexer.py", line 32, in lex_from_str
    from . import getkw
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/getkw.py", line 32, in <module>
    from .atoms import (
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/atoms.py", line 66, in <module>
    bool_t.set_name("bool")
AttributeError: 'Or' object has no attribute 'set_name'. Did you mean: 'setName'?

Traceback (most recent call last):
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/hf_grad_lda/test", line 43, in <module>
    with Path("hf.json").open("r") as o:
  File "/usr/lib64/python3.10/pathlib.py", line 1119, in open
    return self._accessor.open(self, mode, buffering, encoding, errors,
FileNotFoundError: [Errno 2] No such file or directory: 'hf.json'

      Start 23: H_SCF_finite_electric_field
 2/30 Test #23: H_SCF_finite_electric_field ......***Failed    0.32 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h_el_field/h --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h_el_field/h and args None

stdout


stderr
Traceback (most recent call last):
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 90, in <module>
    main()
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 55, in main
    user_dict = parse(infile=inp_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 108, in parse
    ir = lex(infile=infile, ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 40, in lex
    ir = lexer.lex_from_str(in_str=f.read(), ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/lexer.py", line 32, in lex_from_str
    from . import getkw
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/getkw.py", line 32, in <module>
    from .atoms import (
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/atoms.py", line 66, in <module>
    bool_t.set_name("bool")
AttributeError: 'Or' object has no attribute 'set_name'. Did you mean: 'setName'?


      Start 24: H2_SCF_HartreeFock
 3/30 Test #24: H2_SCF_HartreeFock ...............***Failed    0.28 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h2_scf_hf/h2 --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h2_scf_hf/h2 and args None

stdout


stderr
Traceback (most recent call last):
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 90, in <module>
    main()
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 55, in main
    user_dict = parse(infile=inp_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 108, in parse
    ir = lex(infile=infile, ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 40, in lex
    ir = lexer.lex_from_str(in_str=f.read(), ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/lexer.py", line 32, in lex_from_str
    from . import getkw
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/getkw.py", line 32, in <module>
    from .atoms import (
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/atoms.py", line 66, in <module>
    bool_t.set_name("bool")
AttributeError: 'Or' object has no attribute 'set_name'. Did you mean: 'setName'?


      Start 25: H2_static_polarizability_LDA
 4/30 Test #25: H2_static_polarizability_LDA .....***Failed    3.32 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h2_pol_lda/h2 --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x --json
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h2_pol_lda/h2 and args ['--json']

stdout


stderr
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.


      Start 26: H2O_energy_BLYP
 5/30 Test #26: H2O_energy_BLYP ..................***Failed    0.38 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h2o_energy_blyp/h2o --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/h2o_energy_blyp/h2o and args None

stdout


stderr
Traceback (most recent call last):
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 90, in <module>
    main()
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 55, in main
    user_dict = parse(infile=inp_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 108, in parse
    ir = lex(infile=infile, ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 40, in lex
    ir = lexer.lex_from_str(in_str=f.read(), ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/lexer.py", line 32, in lex_from_str
    from . import getkw
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/getkw.py", line 32, in <module>
    from .atoms import (
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/atoms.py", line 66, in <module>
    bool_t.set_name("bool")
AttributeError: 'Or' object has no attribute 'set_name'. Did you mean: 'setName'?


      Start 27: Li_SCF_PBE0
 6/30 Test #27: Li_SCF_PBE0 ......................***Failed    0.26 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/li_scf_pbe0/li --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/li_scf_pbe0/li and args None

stdout


stderr
Traceback (most recent call last):
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 90, in <module>
    main()
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 55, in main
    user_dict = parse(infile=inp_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 108, in parse
    ir = lex(infile=infile, ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 40, in lex
    ir = lexer.lex_from_str(in_str=f.read(), ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/lexer.py", line 32, in lex_from_str
    from . import getkw
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/getkw.py", line 32, in <module>
    from .atoms import (
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/atoms.py", line 66, in <module>
    bool_t.set_name("bool")
AttributeError: 'Or' object has no attribute 'set_name'. Did you mean: 'setName'?


      Start 28: Li_static_polarizability_LDA
 7/30 Test #28: Li_static_polarizability_LDA .....***Failed    0.24 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/li_pol_lda/li --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/li_pol_lda/li and args None

stdout


stderr
Traceback (most recent call last):
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 90, in <module>
    main()
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem", line 55, in main
    user_dict = parse(infile=inp_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 108, in parse
    ir = lex(infile=infile, ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/api.py", line 40, in lex
    ir = lexer.lex_from_str(in_str=f.read(), ir_file=ir_file)
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/lexer.py", line 32, in lex_from_str
    from . import getkw
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/getkw.py", line 32, in <module>
    from .atoms import (
  File "/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/python/mrchem/input_parser/plumbing/atoms.py", line 66, in <module>
    bool_t.set_name("bool")
AttributeError: 'Or' object has no attribute 'set_name'. Did you mean: 'setName'?


      Start 30: Li_solvent_effect
 8/30 Test #30: Li_solvent_effect ................***Failed   24.59 sec

running /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/li_solv/li --executable=/build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/redhat-linux-build/bin/mrchem.x --json
test with input files /build/build/BUILD/mrchem/rawhide/mrchem-1.1.0/tests/li_solv/li and args ['--json']

stdout


stderr
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.


      Start  1: qmfunction
 9/30 Test  #1: qmfunction .......................   Passed    1.56 sec
      Start  2: density
10/30 Test  #2: density ..........................   Passed    0.59 sec
      Start  3: orbital
11/30 Test  #3: orbital ..........................   Passed    0.30 sec
      Start  4: orbital_vector
12/30 Test  #4: orbital_vector ...................Subprocess aborted***Exception:   0.83 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start  5: identity_operator
13/30 Test  #5: identity_operator ................Subprocess aborted***Exception:   1.32 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start  6: position_operator
14/30 Test  #6: position_operator ................Subprocess aborted***Exception:   1.46 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start  7: momentum_operator
15/30 Test  #7: momentum_operator ................Subprocess aborted***Exception:   1.92 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start  8: kinetic_operator
16/30 Test  #8: kinetic_operator .................Subprocess aborted***Exception:   2.11 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start  9: nuclear_operator
17/30 Test  #9: nuclear_operator .................Subprocess aborted***Exception:   2.42 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 10: coulomb_operator
18/30 Test #10: coulomb_operator .................Subprocess aborted***Exception:   4.50 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 11: coulomb_hessian
19/30 Test #11: coulomb_hessian ..................Subprocess aborted***Exception:   3.87 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 12: exchange_operator
20/30 Test #12: exchange_operator ................Subprocess aborted***Exception:   2.94 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 13: exchange_hessian
21/30 Test #13: exchange_hessian .................Subprocess aborted***Exception:   3.55 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 14: xc_operator_lda
22/30 Test #14: xc_operator_lda ..................Subprocess aborted***Exception:   0.98 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 15: xc_operator_blyp
23/30 Test #15: xc_operator_blyp .................Subprocess aborted***Exception:   1.01 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 16: xc_hessian_lda
24/30 Test #16: xc_hessian_lda ...................Subprocess aborted***Exception:   1.01 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 17: xc_hessian_pbe
25/30 Test #17: xc_hessian_pbe ...................Subprocess aborted***Exception:   1.03 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 18: electric_field_operator
26/30 Test #18: electric_field_operator ..........Subprocess aborted***Exception:   2.35 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 19: electric_field_energy
27/30 Test #19: electric_field_energy ............   Passed    0.32 sec
      Start 20: operator_composition
28/30 Test #20: operator_composition .............Subprocess aborted***Exception:   0.89 sec
------------------------------------------------------------
 MPI processes        :    (0 bank)                       1
 OpenMP threads       :                                   8
 Total cores          :                                   8
------------------------------------------------------------
/usr/include/c++/12/bits/stl_vector.h:1122: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = double*; _Alloc = std::allocator<double*>; reference = double*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.

      Start 21: cavity_function
29/30 Test #21: cavity_function ..................   Passed    0.23 sec
      Start 22: reaction_operator
30/30 Test #22: reaction_operator ................   Passed    6.00 sec

20% tests passed, 24 tests failed out of 30

Label Time Summary:
H2O_energy_BLYP                 =   0.38 sec*proc (1 test)
H2_SCF_HartreeFock              =   0.28 sec*proc (1 test)
H2_static_polarizability_LDA    =   3.32 sec*proc (1 test)
HF_molecular_gradient_LDA       =   0.64 sec*proc (1 test)
Li_SCF_PBE0                     =   0.26 sec*proc (1 test)
Li_static_polarizability_LDA    =   0.24 sec*proc (1 test)
cavity_function                 =   0.23 sec*proc (1 test)
coulomb_hessian                 =   3.87 sec*proc (1 test)
coulomb_operator                =   4.50 sec*proc (1 test)
density                         =   0.59 sec*proc (1 test)
dft                             =   0.64 sec*proc (2 tests)
electric_field                  =   0.32 sec*proc (1 test)
electric_field_energy           =   0.32 sec*proc (1 test)
electric_field_operator         =   2.35 sec*proc (1 test)
energy                          =  25.50 sec*proc (4 tests)
exchange_hessian                =   3.55 sec*proc (1 test)
exchange_operator               =   2.94 sec*proc (1 test)
gga                             =   0.38 sec*proc (1 test)
gradient                        =   0.64 sec*proc (1 test)
h2_pol_lda                      =   3.32 sec*proc (1 test)
h2_scf_hf                       =   0.28 sec*proc (1 test)
h2o_energy_blyp                 =   0.38 sec*proc (1 test)
h_el_field                      =   0.32 sec*proc (1 test)
hartree_fock                    =   0.28 sec*proc (1 test)
hf                              =   0.32 sec*proc (1 test)
hf_grad_lda                     =   0.64 sec*proc (1 test)
identity_operator               =   1.32 sec*proc (1 test)
integration                     =  30.02 sec*proc (8 tests)
kinetic_operator                =   2.11 sec*proc (1 test)
li_pol_lda                      =   0.24 sec*proc (1 test)
li_scf_pbe0                     =   0.26 sec*proc (1 test)
li_solv                         =  24.59 sec*proc (1 test)
momentum_operator               =   1.92 sec*proc (1 test)
mrchem                          =  30.02 sec*proc (8 tests)
nuclear_operator                =   2.42 sec*proc (1 test)
open_shell                      =   0.24 sec*proc (1 test)
operator_composition            =   0.89 sec*proc (1 test)
orbital                         =   0.30 sec*proc (1 test)
orbital_vector                  =   0.83 sec*proc (1 test)
polarizability                  =   3.56 sec*proc (2 tests)
position_operator               =   1.46 sec*proc (1 test)
qmfunction                      =   1.56 sec*proc (1 test)
reaction_operator               =   6.00 sec*proc (1 test)
scf                             =  25.45 sec*proc (4 tests)
solvent                         =  24.59 sec*proc (1 test)
unit                            =  41.20 sec*proc (22 tests)
xc_hessian_lda                  =   1.01 sec*proc (1 test)
xc_hessian_pbe                  =   1.03 sec*proc (1 test)
xc_operator_blyp                =   1.01 sec*proc (1 test)
xc_operator_lda                 =   0.98 sec*proc (1 test)

Total Test time (real) =  71.82 sec

The following tests FAILED:
	  4 - orbital_vector (Subprocess aborted)
	  5 - identity_operator (Subprocess aborted)
	  6 - position_operator (Subprocess aborted)
	  7 - momentum_operator (Subprocess aborted)
	  8 - kinetic_operator (Subprocess aborted)
	  9 - nuclear_operator (Subprocess aborted)
	 10 - coulomb_operator (Subprocess aborted)
	 11 - coulomb_hessian (Subprocess aborted)
	 12 - exchange_operator (Subprocess aborted)
	 13 - exchange_hessian (Subprocess aborted)
	 14 - xc_operator_lda (Subprocess aborted)
	 15 - xc_operator_blyp (Subprocess aborted)
	 16 - xc_hessian_lda (Subprocess aborted)
	 17 - xc_hessian_pbe (Subprocess aborted)
	 18 - electric_field_operator (Subprocess aborted)
	 20 - operator_composition (Subprocess aborted)
	 23 - H_SCF_finite_electric_field (Failed)
	 24 - H2_SCF_HartreeFock (Failed)
	 25 - H2_static_polarizability_LDA (Failed)
	 26 - H2O_energy_BLYP (Failed)
	 27 - Li_SCF_PBE0 (Failed)
	 28 - Li_static_polarizability_LDA (Failed)
	 29 - HF_molecular_gradient_LDA (Failed)
	 30 - Li_solvent_effect (Failed)
Errors while running CTest
error: Bad exit status from /var/tmp/rpm-tmp.dRpdrH (%check)
```

**Comment by susilehtola** [comment]
This appears to be related to changes in pyparsing api
https://readthedocs.org/projects/pyparsing-docs/downloads/pdf/latest/

**Comment by susilehtola** [comment]
The culprit is commit c752bdbd419ba75c4aa2b279c43a279cd0bd2ba7

**Comment by susilehtola** [comment]
The calls have been changed to pyparsing 3 syntax for no reason, since the changelog states
> camelCase names have been converted to PEP-8 snake_case names. Method names and arguments that were camel case (such as parseString) have been replaced with PEP-8 snake case versions (parse_string). Backward-compatibility synonyms for all names and arguments have been included, to allow parsers written using the old names to run without change.  The synonyms will be removed in a future release. New parser code should be written using the new PEP-8 snake case names

**Comment by susilehtola** [comment]
> MRChem bundles its own copy of `pyparsing`. The bundling happens when we regenerate the parsing infrastructure with `parselglossy`. Sadly, there was a tiny bug in the bundling, which has slipped through. The fix should propagate in a few days and Stig can mint 1.1.1

Why though? pyparsing is a standard package. Bundling should not be necessary; at the very least, it should be possible to turn it off...

**Comment by robertodr** [comment]
Yes, it's a commonly used package, but it's not in the standard Python distribution. We made the choice to bundle to avoid having to install Python packages. The import statements will prefer system-installed pyparsing over the bundled one, if it's pyparsing>=3

**Comment by susilehtola** [comment]
Can the 2.x syntax not be restored?

**Comment by stigrj** [comment]
Sorry for my absence. @robertodr is the fix ready? What exactly needs to be done on the MRChem side?

**Comment by robertodr** [comment]
Fix is ready, but will require a new release (unless you want to edit the tag for 1.1.0) The attached patch restore 2.x syntax and I guess it can be applied during the Fedora build. 

[pp-patch.txt](https://github.com/MRChemSoft/mrchem/files/9695671/pp-patch.txt)

**Comment by stigrj** [comment]
Great, thanks! I will make a new release including a fix for #426, hopefully by tomorrow.

**Comment by stigrj** [comment]
Sorry for the delay, but now an official fix is finally out (v1.1.1). Closing this issue, but please re-open if the problem persists.
